simulator/func_sim/rf/t/unit_test.cpp

```diff
@@ -32,21 +32,49 @@ static_assert(MIPSRegister::MAX_REG >= 32);
 TEST( RF, read_write_rf)
 {
     auto rf = std::make_unique<TestRF>();
-    
+
     // Fill array using write() and check correctness using read()
     for( size_t i = 0; i < 32; ++i)
-    {   
+    {
         rf->write( MIPSRegister(i), i);
-        
+
         // Try to write something in zero register
         rf->write( MIPSRegister::zero, i);
-        
+
         // Checks
         ASSERT_EQ( rf->read( MIPSRegister(i)), i);
         ASSERT_EQ( rf->read( MIPSRegister::zero), 0u);
+
+        // Check instructions which need mask
+        rf->write( MIPSRegister(i), i, 0x0);
```

Comment:
These tests can be easily broken, as you write the same value as it was before. Actually if you did no mask handling inside, your tests would pass with 100% rate. Our goal is completely different, we would like to prevent making errors in mask handling.

Could you do something more complicated:
```c++
rf->write( MIPSRegister(i), 0);
rf->write( MIPSRegister(i), 0x12345678, 0xFF);
ASSERT_EQ( rf->read( MIPSRegister(i)), 0x78); // only one byte must go

rf->write( MIPSRegister(i), 0);
rf->write( MIPSRegister(i), 0x12345678, 0xFF00);
ASSERT_EQ( rf->read( MIPSRegister(i)), 0x5600); // only one byte must go
// ...
```